networkx/utils/backends.py

```diff
@@ -895,8 +938,44 @@ def _convert_and_call_for_tests(
                     G2._succ.clear()
                     G2._succ.update(G1._succ)
                 return G2
-
-        return backend.convert_to_nx(result, name=self.name)
+            return backend.convert_to_nx(result)
+
+        converted_result = backend.convert_to_nx(result)
+        if isinstance(converted_result, nx.Graph) and self.name not in {
+            "boykov_kolmogorov",
+            "preflow_push",
+            "quotient_graph",
+            "shortest_augmenting_path",
+            "spectral_graph_forge",
+            "star_graph",  # Mutates input list (appends n)
```

Comment:
Should we update `star_graph` so it doesn't mutate the input argument?
Change `nodes.append(n)` to e.g. `nodes = [*nodes, n]`

Reply:
`star_graph` does not mutate the input.  It should be OK as is.
The input `n` is (or should be) processed by the `nodes_or_number` decorator and becomes `n, nodes` where (when `n` isinstance `Integral`) `nodes` is a list of nodes that can safely be appended to.

Reply:
Ah, you are correct. Then we should be able to remove `"star_graph"` from this set. Let me try! I think I know why I was seeing an issue before, and it was something _I_ was doing in a backend.

Reply:
I see. CI failed b/c `nodes_or_number` is processed _before_ dispatching, so each call to `star_graph` mutates the list returned by `nodes_or_number`. Hence, user input isn't mutated, but dispatch input _is_. Perhaps `nodes_or_number` should go _after_ `_dispatch` (or we could change `star_graph` to not mutate the normally-safe-to-mutate nodes list).

Reply:
What is the problem with the dispatch input being mutated? Is it that it **has** to be mutated to match what networkx is doing? and that same code is also used somewhere else where you don't want to mutate the input? 

I'm just trying to ask whether this is something we shouldn't do because of style, or something we shouldn't do because it messes up other user types, etc.

Reply:
I will also say that backend developers might be confused that input `n` is a 2-tuple with information: `(n, nodes)`. Perhaps we should put nodes_and_number *after* dispatch. But then the backend would have to code up handling both an int or a container of nodes. I think you may have more intuition than anyone about which is better.  I'd rather avoid having to construct a new list in case it is a million nodes long.

Reply:
My latest commit moved `@nodes_or_number` before `@nx._dispatch` to avoid this. Now, backends will get the original argument, and if they want to use `@nodes_or_number`, they will need to do so themselves. This is fine for the backends I work on.